bufferserver/src/main/java/com/datatorrent/bufferserver/internal/LogicalNode.java

```diff
@@ -115,12 +115,7 @@ public void addConnection(WriteOnlyClient connection)
    */
   public void removeChannel(WriteOnlyClient client)
   {
-    for (PhysicalNode pn : physicalNodes) {
-      if (pn.getClient() == client) {
-        physicalNodes.remove(pn);
-        break;
-      }
-    }
+    physicalNodes.removeIf(node -> (node.getClient() == client));
```

Comment:
I can do that but wouldn't you want a stricter check here. Also equals does not seem to be implemented/overridden in the client implementation, should we still go with equals

Reply:
I prefer `equals` as it delegates to the `WriteOnlyClient` to define equality rather than hardcoding reference equality into the `LogicalNode`.

Reply:
Normally I would think it would be the decision of LogicalNode but I also see that in the add it is relying on equals so it would be ok to delegate in remove as well.